##  Task (1)
flag{s0m3_susp1c10us_str1ng}

![t1cttr](https://user-images.githubusercontent.com/123714177/230892539-9dbcbedc-8c28-41cc-b870-1bb4a7cc47b5.png)
![t1cbrchf](https://user-images.githubusercontent.com/123714177/230892565-acebcd9f-5717-4923-99ea-c6b811af9caa.png)


##  Task (2)
flag{sup3r_s1mpl3_x0r}

I opened the task in clutter. There was a prompt "Enter the first magic number" and "hmm" function was called.
![t2main](https://user-images.githubusercontent.com/123714177/230893587-fba55857-6582-4b52-a952-0fb4345c21a5.png)

Then, I went to analyze "hmm" function. There were if conditions which were validating the input and upon giving the correct input, it was asking for the next magic number. Correct input for 
1st magic number = 0x17(hex)  = 23(dec)
2nd magic number = 0x539(hex) = 1337(dec)
3rd magic number = 0xfc(hex)  = 252(dec)
![t2hmm](https://user-images.githubusercontent.com/123714177/230894783-45eedfbb-62b9-45a2-b152-4686f176b411.png)

Then, I run the task2 in terminal and upon entering correct inputs, got the flag.
![t2flag](https://user-images.githubusercontent.com/123714177/230895009-00a42e74-b3ad-456d-a928-ddd8d31c5615.png)


##  Task (3)
"flag{r3v3rs3_3ng1n33r1ng_101}"


##  Task (4)
flag{4_m3d10cr3_m4lw4r3_ch4ll3nge}
